tags/TagContext.md

```diff
@@ -13,11 +13,23 @@ all of the information that must be propagated across process boundaries.
 
 ## TagKey
 
-A string or string wrapper, with some restrictions:
+A TagKey consists of a Name and a Scope.
```

Comment:
I would rather not specify that the scope is an attribute of TagKey.  It may make sense to support tag scope in some implementations by providing a way to "register" a tag for propagation, with a function that takes a TagKey and a Scope.  For example, it could work like this in Java:

```java
static final TagKey MY_KEY = ...
static final TagManager tagManager = ...

static initialize() {
  ...
  tagManager.registerTagScope(MY_KEY, TagScope.REQUEST);
```

This API is more flexible because it separates creating tag keys from setting their scope, and it avoids adding scope information to views, which contain TagKeys but don't need to specify how the tags are propagated.

Reply:
fyi there's some discussion here.. I do like the idea of "request" and "local" scopes https://github.com/openzipkin/brave/pull/577

Also, I'm not sure, but it does feel like a registration approach could be helpful. I wonder if we shouldn't start with an experimental api before doing a spec. At least before merging a spec